timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/server/healthchecks/databasechecks/SortIndexesDatabaseCheck.java

```diff
@@ -0,0 +1,67 @@
+package nl.knaw.huygens.timbuctoo.server.healthchecks.databasechecks;
+
+import javaslang.control.Try;
+import nl.knaw.huygens.timbuctoo.search.description.IndexDescription;
+import nl.knaw.huygens.timbuctoo.search.description.indexes.IndexDescriptionFactory;
+import nl.knaw.huygens.timbuctoo.search.description.indexes.IndexerSortFieldDescription;
+import nl.knaw.huygens.timbuctoo.server.healthchecks.DatabaseCheck;
+import nl.knaw.huygens.timbuctoo.server.healthchecks.ElementValidationResult;
+import nl.knaw.huygens.timbuctoo.server.healthchecks.ValidationResult;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.structure.VertexProperty;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static nl.knaw.huygens.timbuctoo.model.GraphReadUtils.getEntityTypes;
+import static nl.knaw.huygens.timbuctoo.model.GraphReadUtils.getProp;
+
+public abstract class SortIndexesDatabaseCheck implements DatabaseCheck {
+
+  protected ValidationResult getValidationResultForType(Vertex vertex, String type) {
+    List<String> types = Arrays.asList(getEntityTypes(vertex)
+            .orElseGet(() -> Try.success(new String[0]))
+            .getOrElse(() -> new String[0]));
+
+    if (types.contains(type)) {
```

Comment:
als je getIndexersForType gebruikt heb je de subclasses niet nodig en hebben we gewoon 1 SortIndexesDatabaseCheck die niet hoeft te worden aangepast als je een nieuwe databaseCheck aanmaakt